lib/web_ui/lib/src/engine/text_editing/text_editing.dart

```diff
@@ -147,19 +143,53 @@ void _styleAutofillElements(
   elementStyle.setProperty('caret-color', 'transparent');
 }
 
+void _ensureEditingElementInView(DomElement element, int viewId) {
+  final bool isAlreadyAppended = element.isConnected ?? false;
+  if (!isAlreadyAppended) {
+    // If the element is not already appended to a view, we don't need to move
+    // it anywhere.
+    return;
+  }
```

Comment:
I understand where you are coming from, but I don't think that will work. Let me try to explain..

**Why check for `isConnected`?**

The way our text editing system works is we don't append the input element until we receive a `TextInput.show` message from the framework. Before that point, the framework could send us any number of messages about the text field (e.g. `setClient`, `updateConfig`, etc).

We NEED to make sure that we don't accidentally append the input to the DOM before it's supposed to. That's why the code checks for `isConnected` first to see if the input has already been appended or not. If it's not in the DOM yet, that means we haven't receive `TextInput.show` and we SHOULDN'T append it (later, when `TextInput.show` comes, the input will be appended to the most up-to-date `viewId`).

**Why check for `currentView.viewId != viewId`?**

As you said, if we don't do this check, the input element will be re-appended to the DOM. Even if it remains in the same view, the state is going to be reset (see [jsfiddle](https://jsfiddle.net/fm9xcyw4/1/)).

Reply:
OK, this is needed to make the call idempotent and to not do work unless really necessary, because every time we move the text editing input, we reset its state (wowza)